app/controllers/v0/sign_in_controller.rb

```diff
@@ -179,108 +193,68 @@ def validate_token_params(code, code_verifier, grant_type)
+      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_AUTHORIZE_ATTEMPT_FAILURE,
+                       tags: ["type:#{type}, client_id:#{client_id}, acr:#{acr}"])
       render json: { errors: error }, status: :bad_request
     end
 
-    def handle_token_error(error)
-      context = { code: params[:code], code_verifier: params[:code_verifier], grant_type: params[:grant_type] }
+    def handle_callback_error(error, state_payload, state, code)
+      type = state_payload&.type
+      client_id = state_payload&.client_id
+      acr = state_payload&.acr
+      context = { type: type, client_id: client_id, acr: acr, state: state, code: code }
       log_message_to_sentry(error.message, :error, context)
-      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_TOKEN_FAILURE)
+      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_CALLBACK_FAILURE,
+                       tags: ["type:#{type}, client_id:#{client_id}, acr:#{acr}"])
       render json: { errors: error }, status: :bad_request
     end
 
-    def handle_refresh_error(error, status: :unauthorized)
-      context = { refresh_token: params[:refresh_token], anti_csrf_token: params[:anti_csrf_token] }
-      log_message_to_sentry(error.message, :error, context)
-      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_REFRESH_FAILURE)
-      render json: { errors: error }, status: status
-    end
-
-    def handle_revoke_error(error, status: :unauthorized)
-      context = { refresh_token: params[:refresh_token], anti_csrf_token: params[:anti_csrf_token] }
+    def handle_post_login_error(error, context, statsd_code, user_id = nil, status: :unauthorized)
+      auth_info = user_id ? get_user_auth_info(user_id) : {}
+      context = context.merge({ type: auth_info[:type], client_id: auth_info[:client_id], loa: auth_info[:loa] })
       log_message_to_sentry(error.message, :error, context)
-      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_REVOKE_FAILURE)
+      StatsD.increment(statsd_code,
+                       tags: ["type:#{auth_info[:type]}, client_id:#{auth_info[:client_id]}, loa:#{auth_info[:loa]}"])
       render json: { errors: error }, status: status
     end
 
-    def handle_introspect_error(error)
-      context = { user_uuid: @current_user.uuid }
-      log_message_to_sentry(error.message, :error, context)
-      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_INTROSPECT_FAILURE)
-      render json: { errors: error }, status: :unauthorized
-    end
-
-    def handle_logout_error(error)
-      context = { user_uuid: @current_user.uuid }
-      log_message_to_sentry(error.message, :error, context)
-      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_LOGOUT_FAILURE)
-      render json: { errors: error }, status: :unauthorized
-    end
-
-    def handle_revoke_all_sessions_error(error)
-      context = { user_uuid: @current_user.uuid }
+    def handle_authenticated_route_error(error, statsd_code)
```

Comment:
Nice work refactoring all these logging functions! Hopefully it'll be easier to parse the sign in controller after all this